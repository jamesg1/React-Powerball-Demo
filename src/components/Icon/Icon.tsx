import React from 'react';
import Lightning from './Icons/Lightning';
import Trash from './Icons/Trash';
import Cross from './Icons/Cross';

export interface IconProps {
  width?: number | string;
  fill?: string;
  name: 'lightning' | 'trash' | 'cross';
}

const Icon: React.FC<IconProps> = (props: IconProps) => {
  switch (props.name) {
    case 'lightning':
      return <Lightning {...props} />;
    case 'trash':
      return <Trash {...props} />;
    case 'cross':
      return <Cross {...props} />;
    default:
      return null;
  }
};

Icon.defaultProps = {
  width: '100%',
  fill: '#ffffff'
};

export default Icon;
