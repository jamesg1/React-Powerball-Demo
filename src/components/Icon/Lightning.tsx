import React from 'react';
import { IconProps } from './Icon';

const Lightning: React.FC<IconProps> = ({ width, fill }: IconProps) => (
  <svg
    width={width}
    height={width}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink">
    {/* Source: https://iconmonstr.com/connection-1-svg/ */}
    <path fill={fill} d="M8 24l3-9h-9l14-15-3 9h9l-14 15z" />
  </svg>
);

export default Lightning;
