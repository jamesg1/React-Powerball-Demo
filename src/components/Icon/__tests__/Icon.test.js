import React from 'react';
import Icon from '../Icon';
import renderWithTheme from 'styling/withTheme';
import 'jest-styled-components';

describe('<Icon />', () => {
  const lightning = 'lightning';
  const trash = 'trash';
  const cross = 'cross';

  test('it renders correct icon svg based on name prop', () => {
    let tree = renderWithTheme(<Icon name={lightning} />).toJSON();
    expect(tree.props.id).toEqual(`${lightning}-icon`);
    expect(tree).toMatchSnapshot();
    tree = renderWithTheme(<Icon name={trash} />).toJSON();
    expect(tree.props.id).toEqual(`${trash}-icon`);
    expect(tree).toMatchSnapshot();
    tree = renderWithTheme(<Icon name={cross} />).toJSON();
    expect(tree.props.id).toEqual(`${cross}-icon`);
    expect(tree).toMatchSnapshot();
  });

  test('it renders null when name is not defined', () => {
    const tree = renderWithTheme(<Icon />).toJSON();
    expect(tree).toBeNull();
  });

  test('it changes size and color fill based on props', () => {
    const width = '25px';
    const fill = '#DDDDDD';
    const tree = renderWithTheme(<Icon name={lightning} width={width} fill={fill} />).toJSON();
    expect(tree.props.width).toEqual(width);
    expect(tree.props.height).toEqual(width);
    expect(tree).toMatchSnapshot();
  });
});
