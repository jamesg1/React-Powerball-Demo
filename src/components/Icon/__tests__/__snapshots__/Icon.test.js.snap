// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> it changes size and color fill based on props 1`] = `
<svg
  height="25px"
  id="lightning-icon"
  viewBox="0 0 24 24"
  width="25px"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <path
    d="M8 24l3-9h-9l14-15-3 9h9l-14 15z"
    fill="#DDDDDD"
  />
</svg>
`;

exports[`<Icon /> it renders correct icon svg based on name prop 1`] = `
<svg
  height="100%"
  id="lightning-icon"
  viewBox="0 0 24 24"
  width="100%"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <path
    d="M8 24l3-9h-9l14-15-3 9h9l-14 15z"
    fill="#ffffff"
  />
</svg>
`;

exports[`<Icon /> it renders correct icon svg based on name prop 2`] = `
<svg
  height="100%"
  id="trash-icon"
  viewBox="0 0 24 24"
  width="100%"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <path
    d="M3 6v18h18v-18h-18zm5 14c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 1-1s1 .448 1 1v10zm5 0c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 1-1s1 .448 1 1v10zm5 0c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 1-1s1 .448 1 1v10zm4-18v2h-20v-2h5.711c.9 0 1.631-1.099 1.631-2h5.315c0 .901.73 2 1.631 2h5.712z"
    fill="#ffffff"
  />
</svg>
`;

exports[`<Icon /> it renders correct icon svg based on name prop 3`] = `
<svg
  height="100%"
  id="cross-icon"
  viewBox="0 0 24 24"
  width="100%"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <path
    d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
    fill="#ffffff"
  />
</svg>
`;
